### PR TITLE
Login Servlet

### LINKED ISSUE #32: Week 4 Project: Authentication - Step - 2
Add a servlet that returns the login status of the user.
* Test that this works by running a dev server and navigating to the servlet's URL.
* When you get this step working, create a pull request and send it to your host for review!

### PR DESCRIPTION
**Pull request by SibingWu**
## Description

Change the file structure.
Add a servlet that returns the login status of the user.

Fixes #32 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Test
Manual Test by running a dev server and navigating to the servlet's URL.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/authentication/servlets/LoginServlet.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import com.google.appengine.api.users.UserServiceFactory;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
```
Comment (htthach) [inline portfolio/src/main/java/com/google/sps/authentication/servlets/LoginServlet.java]: Let's make it final class. And also add some public javadoc.

File: portfolio/src/main/java/com/google/sps/authentication/servlets/LoginServlet.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    private UserService userService;
+
+    @Override
+    public void init() {
+        this.userService = UserServiceFactory.getUserService();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (htthach) [inline portfolio/src/main/java/com/google/sps/authentication/servlets/LoginServlet.java:25]: Does the tutorial suggest using get or something else? How would the password be transmitted?

File: portfolio/src/main/java/com/google/sps/authentication/servlets/LoginServlet.java (hunk 1 above)
Comment (SibingWu) [reply portfolio/src/main/java/com/google/sps/authentication/servlets/LoginServlet.java]: Fixed

File: portfolio/src/main/java/com/google/sps/authentication/servlets/LoginServlet.java (hunk 2 above)
Comment (SibingWu) [reply portfolio/src/main/java/com/google/sps/authentication/servlets/LoginServlet.java:25]: hmmm the tutorial does not mention... actually when direct to the log in page, only email address is needed, no place for password

Comment (SibingWu) [reply portfolio/src/main/java/com/google/sps/authentication/servlets/LoginServlet.java:25]: The example use GET

Comment (SibingWu) [reply portfolio/src/main/java/com/google/sps/authentication/servlets/LoginServlet.java:25]: Fixed with adding comments as discussed offline

**Review by htthach — APPROVED** [review]

**Pull request merged**